homeassistant/components/daikin/config_flow.py

```diff
@@ -124,3 +124,11 @@ async def async_step_discovery(self, discovery_info):
         self._abort_if_unique_id_configured()
         self.host = discovery_info[KEY_IP]
         return await self.async_step_user()
+
+    async def async_step_zeroconf(self, discovery_info):
+        """Prepare configuration for a discovered Daikin device."""
+        _LOGGER.debug("Zeroconf discovery_info: %s", discovery_info)
+        await self.async_set_unique_id(discovery_info.get(CONF_HOST))
```

Comment:
It looks like you are using the `mac` for the unique id above 👍  `await self.async_set_unique_id(mac)`

Is the mac address in the `discovery_info` ?

Reply:
> Is the mac address in the discovery_info ?

No it is not, that is why we are using `CONF_HOST`..

Reply:
If the host is an ip address it may change. We don't want to use ip addresses for unique id.

Reply:
> If the host is an ip address it may change. We don't want to use ip addresses for unique id.

So how should I approach it?

Reply:
Use something that is stable and unique to identify the device from the discovery info.

Reply:
This is the only thing available: 
```
{'host': '10.0.0.6', 'port': 80, 'hostname': 'DaikinAP34738.local.', 'type': '_dkapi._tcp.local.', 'name': 'DkApi._dkapi._tcp.local.', 'properties': {'_raw': {'reg': b'eu', 'type': b'aircon', 'ver': b'1_2_51', 'adp_kind': b'3'}, 'reg': 'eu', 'type': 'aircon', 'ver': '1_2_51', 'adp_kind': '3'}}
```

Reply:
You know the integration and device best.

Reply:
Does `DaikinAP34738` remain the same when the ip changes?

Reply:
I don't know, my Daikin unit doesn't support zeroconf.

Reply:
@TopdRob what is your take on this?

Reply:
Yes it stays unique. It's the SSID of the access point mode.

Reply:
Since the user can't change this, seems stable enough to me.

Reply:
We didn't change to `"hostname"` key before merge.

Reply:
@MartinHjelmare Good catch.  Got lost in the details and missed that "host" != "hostname"   @fredrike Would you please send a new PR to adjust.